crates/server/src/authority/authority.rs

```diff
@@ -201,3 +198,103 @@ pub trait DnssecAuthority: Authority {
     /// Sign the zone for DNSSEC
     async fn secure_zone(&self) -> DnsSecResult<()>;
 }
+
+/// Result of a Lookup in the Catalog and Authority
+pub enum LookupResult<T, E = LookupError> {
```

Comment:
Do we actually need to abstract over `E`, or could we just force this to be `LookupError` for now?

Reply:
Not if we want to provide a map_err implementation.  Having the Err variant default to LookupError but support a generic type seemed like the best compromise to me.